Mathlib/Combinatorics/SimpleGraph/Basic.lean

```diff
@@ -1299,23 +1349,69 @@ theorem map_adj (f : V ↪ W) (G : SimpleGraph V) (u v : W) :
   Iff.rfl
 #align simple_graph.map_adj SimpleGraph.map_adj
 
+theorem map_adj_apply {G : SimpleGraph V} {f : V ↪ W} {a b : V} :
+    (G.map f).Adj (f a) (f b) ↔ G.Adj a b := by simp
+#align simple_graph.map_adj_apply SimpleGraph.map_adj_apply
+
+@[simp]
+theorem map_id : G.map (Function.Embedding.refl _) = G :=
+  SimpleGraph.ext _ _ <| Relation.map_id_id _
+#align simple_graph.map_id SimpleGraph.map_id
+
+@[simp]
+theorem map_map (f : V ↪ W) (g : W ↪ X) : (G.map f).map g = G.map (f.trans g) :=
+  SimpleGraph.ext _ _ <| Relation.map_map _ _ _ _ _
+#align simple_graph.map_map SimpleGraph.map_map
+
 theorem map_monotone (f : V ↪ W) : Monotone (SimpleGraph.map f) := by
   rintro G G' h _ _ ⟨u, v, ha, rfl, rfl⟩
   exact ⟨_, _, h ha, rfl, rfl⟩
 #align simple_graph.map_monotone SimpleGraph.map_monotone
 
+instance decidableMap (f : V ↪ W) (G : SimpleGraph V) [DecidableRel (Relation.Map G.Adj f f)] :
+    DecidableRel (G.map f).Adj :=
+  ‹DecidableRel _›
+#align simple_graph.decidable_map SimpleGraph.decidableMap
+
 /-- Given a function, there is a contravariant induced map on graphs by pulling back the
 adjacency relation.
 This is one of the ways of creating induced graphs. See `SimpleGraph.induce` for a wrapper.
 
 This is surjective when `f` is injective (see `SimpleGraph.comap_surjective`).-/
-@[simps]
```

Comment:
Why did this change?

Reply:
The generated lemma `comap_Adj` does not follow the naming convention.

Reply:
Please teach simps to generate the correct name

Reply:
Actually, the generated `comap_adj` has explicit arguments that should be implicit. So I'm still removing the `simps` even though the generated lemma now has the correct name.

Reply:
Can you file an issue about `simps` doing that?

Reply:
#8275